qutip/solver/result.py

```diff
@@ -602,21 +645,25 @@ def _post_init(self):
         self.num_trajectories = 0
         self._target_ntraj = None
 
-        store_states = self.options['store_states']
-        store_final_state = self.options['store_final_state']
-        store_traj = self.options['keep_runs_results']
+        store_density_matricies = self.options['store_density_matricies']
+        store_final_matrix = self.options['store_final_density_matrix']
+        store_trajectory = (self.options['keep_runs_results']
+                            or self.options['store_states'])
+        store_final_state = self.options["store_final_state"]
```

Comment:
If `self.options['store_states']` is `False` or `None` and expectation are computed, each trajectories will not save the states. Therefore `trajectory.states` will be None and the average state can't be computed.

Same for `store_final_matrix`, it can't be computed if each state does not store the it's final state.

Reply:
Ah ok thanks, so I'll need to make sure the trajectory options are also modified if `store_final_density_matrix = True`. I'll take a look and fix all of the tests on Thursday